charts/ColorScale.ts

```diff
@@ -166,20 +171,43 @@ export class ColorScale {
     @computed get autoBinMaximums(): number[] {
         if (!this.sortedNumericValues.length || this.numAutoBins <= 0) return []
 
-        const { binStepSize, numAutoBins, minBinValue } = this
-
-        const bucketMaximums = []
-        let nextMaximum = minBinValue + binStepSize
-        for (let i = 0; i < numAutoBins; i++) {
-            bucketMaximums.push(nextMaximum)
-            nextMaximum += binStepSize
+        if (this.config.binningStrategy === ColorScaleBinningStrategy.ckmeans) {
+            const clusters = ckmeans(
+                this.sortedNumericBinningValues,
+                this.numAutoBins
+            )
+            // The output of this algorithm can create bins that start & end at the same value.
+            // This also means the first bin can both start and end at the same value – the minimum
+            // value. This is why we uniq() and why we remove any values <= minimum value.
+            return excludeUndefined(uniq(clusters.map(last))).filter(
+                v => v > this.autoMinBinValue
+            )
+        } else if (
+            this.config.binningStrategy === ColorScaleBinningStrategy.quantiles
+        ) {
+            return excludeUndefined(
+                range(1, this.numAutoBins + 1).map(v =>
+                    quantile(
+                        this.sortedNumericBinningValues,
+                        v / this.numAutoBins
+                    )
+                )
+            )
+        } else {
+            // Equal-interval strategy by default
+            const { binStepSize, numAutoBins, minBinValue } = this
+            const bucketMaximums = []
+            let nextMaximum = minBinValue + binStepSize
+            for (let i = 0; i < numAutoBins; i++) {
+                bucketMaximums.push(nextMaximum)
+                nextMaximum += binStepSize
+            }
+            return bucketMaximums
```

Comment:
I know this is old code, but what about rewriting this like so:

```suggestion
            return range(1, numAutoBins + 1).map(v => 
            	minBinValue + v * binStepSize
            )
```

Reply:
Much more elegant!